File: engine/model/src/main/scala/tools/packet/npctalk/GetNPCTalkText.scala
package tools.packet.npctalk

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GetNPCTalkText(private var _npcId: Int, private var _talk: String, private var _def: String) extends PacketInput {
  def npcId: Int = _npcId

  def talk: String = _talk

  def theDef: String = _def

  override def opcode(): SendOpcode = SendOpcode.NPC_TALK
}

File: engine/engine-model/src/main/java/tools/packet/npctalk/GetNPCTalkText.java
package tools.packet.npctalk;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GetNPCTalkText(Integer npcId, String talk, String theDef) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.NPC_TALK;
   }
}